src/editmap.cpp

```diff
@@ -1831,6 +1831,129 @@ int editmap::mapgen_preview( real_coords &tc, uimenu &gmenu )
     return ret;
 }
 
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////
+/*
+ * Display mapgen results over selected target position, and optionally regenerate / apply / abort
+ */
+int editmap::mapgen_set( real_coords &tc, std::string om_name)
```

Comment:
That content of this function looks suspiciously like it's just copied code fragments from several places in this file. Please don't just copy code. If you need to run the same code from several places, move it into a separate function and call that function from those places.